server/plugins/encryption/wrapper/store/secret_store_wrapper.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2022 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/woodpecker-ci/woodpecker/server/model"
+)
+
+type EncryptedSecretStore struct {
+	store      model.SecretStore
+	encryption model.EncryptionService
+}
```

Comment:
```suggestion
}

// ensure wrapper match interface
var _ model.SecretStore = new(EncryptedSecretStore)
```

just a nit to make sure thing work, and you will get an error in IDe asap you miss something or on compiletime afterwards ...